tests/unit/components/playback/controls/InputDropDown.spec.js

```diff
@@ -0,0 +1,218 @@
+    // note the store will mutate across tests, so make sure to re-create it in beforeEach
+    const storePath = `${process.env.buildDir}/store.js`;
+    NuxtStore = await import(storePath);
+  });
+
+  beforeEach(async () => {
+    store = await NuxtStore.createStore();
+  });
+
+  afterEach(() => wrapper.destroy());
+  const nicknames = [
+    "Customer Account 1",
+    "Customer Account 2",
+    "Customer Account 3",
+  ];
+  test("When mounting InputDropDown from the build dist file, Then it loads successfully and the props defined title is rendered", () => {
+    const propsData = {
+      title_label: "Customer ID",
+      options_text: nicknames,
+    };
+    wrapper = shallowMount(DistComponentToTest, {
+      propsData,
+      store,
+      localVue,
+    });
+    const target_span = wrapper.find(".span__input-dropdown-content-label");
+
+    expect(target_span.text()).toStrictEqual("Customer ID");
+  });
+  test("When the InputDropDown is mounted, Then it loads successfully and the props defined placeholder is rendered on input field", () => {
+    const propsData = {
+      title_label: "Customer ID",
+      options_text: nicknames,
+      placeholder: "Select Customer ID",
+    };
+    wrapper = mount(ComponentToTest, {
+      propsData,
+      store,
+      localVue,
+    });
+    const target_input = wrapper.find("#input-dropdown-widget");
+    expect(target_input.attributes().placeholder).toStrictEqual(
+      "Select Customer ID"
+    );
+  });
+  test("When the InputDropDown is mounted, Then the user enters few charters in the input, confirm that an event 'update' is emitted with entered text", async () => {
+    const propsData = {
+      title_label: "Customer ID",
+      options_text: nicknames,
+      placeholder: "Select Customer ID",
+      value: "",
+    };
+    wrapper = mount(ComponentToTest, {
+      propsData,
+      store,
+      localVue,
+    });
+    const customer = "New Customer"; // new Customer
+    const input_widget = wrapper.find("#input-dropdown-widget");
+    input_widget.element.value = customer;
+    await input_widget.trigger("input");
+    const parent_id_events = wrapper.emitted("update:value");
+    expect(parent_id_events).toHaveLength(1);
+    expect(parent_id_events).toStrictEqual([[customer]]);
+  });
+  test("When the InputDropDown is mouted, Then a prop message_if_blank is set to true and invalid_text is provided as value is <empty>, an error text is rendered", async () => {
+    const propsData = {
+      title_label: "Customer ID",
+      options_text: nicknames,
+      placeholder: "Select Customer ID",
+      value: "",
+      invalid_text: "This field is required",
+      message_if_blank: true,
+    };
+    wrapper = mount(ComponentToTest, {
+      propsData,
+      store,
+      localVue,
+    });
+    const target_div = wrapper.find(
+      ".div__input-dropdown-controls-content-feedback"
+    );
+    expect(target_div.text()).toStrictEqual("This field is required");
```

Comment:
It looks like if the field is optional (`message_if_blank=false`), then the error does not show up if a user types in an invalid choice.  Please create a failing unit test replicating this issue and then correct the problem in the code.
![image](https://user-images.githubusercontent.com/5436635/97766307-24fb8700-1aec-11eb-958d-54ae833431cb.png)

Reply:
NO don't agree, this is not problem, if some one wants to have the field message_if_black=false, he is forcing the system to not give feedback,  the logic of validation is under the parent or the user of this component. 

Incase , if there is no feedback to provide then message_if_black should be set to true and the invalid_text be blank.

When some one enter "**boetheant**" the parent has to set the value of message_if_black=false on then response receiving event function, has to run the logic of validation and set the message_if_black=true and give the feedback text in the invalid_text reason, this is not a part of InputDropDown Widget functionality.

JEST UNIT testcase can't have a logic of the parent validation brought into the widget verification or testing, this is not feasible.

Just like all the validation or managed in the parent for **InputWidget**, same happens in **InputDropDown**. 
So both have identical control, this is a wrong observation which cannot be implemented, in the InputDropDown widget.

Reply:
Agreed---we can have this component be very flexible by not including any validation logic at all.

However, we have two very common use cases for the validation logic of a dropdown widget, so in the near future we should create a parent widget to make the use cases easy to code:

In almost all cases, we want something to be considered invalid if it does not match a choice from the list. We would use the same invalid message for any situation (except blank) where the user input does not match a choice in the list, the input would be valid if it matched any choice in the list.  So something with props like: `message_when_not_matching_list`, and `message_when_blank` (which could be an empty string indicating that the field is optional and blank is valid). And then that component takes care of that simple and very common validation logic